packages/SwingSet/test/promise-watcher/bootstrap-promise-watcher.js

```diff
@@ -0,0 +1,45 @@
+import { E } from '@endo/eventual-send';
+import { Far } from '@endo/marshal';
+import { makePromiseKit } from '@endo/promise-kit';
+
+export function buildRootObject() {
+  let vatAdmin;
+  let uptonRoot;
+  let uptonAdmin;
+  const pk1 = makePromiseKit();
+  const pk2 = makePromiseKit();
+  const pk3 = makePromiseKit();
+  const pk4 = makePromiseKit();
+
```

Comment:
this test is great! I think we should exercise a few additional things:
* result promises: have vat-bootstrap create another two PromiseKits, and create a pair of `Far` objects which return the `.promise` (one each). Then inside `upgradeV2`, resolve one and reject the other. Create a second pair of `Far` objects, one of which returns `val5` and the other does `throw('val6')` or something. The vat under test should `VatData.watch(E(far1).foo())` for all four: two should resolve/reject promptly, the other two will fire after upgrade, and the watcher should see it in version-2.
* let's exercise the multiple argument ability: `VatData.watch(p, watcher, 'arg1', 'arg2')`, where we're currently only passing a single argument
* the test that exercises the `makeDK` option should exercise the omission of `onFulfilled` and/or `onRejected`, to demonstrate that it doesn't crash anything (unless that runs afoul of an aggressive unhandled-rejection checker)